opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -227,7 +227,19 @@ namespace Opm {
 
         using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
+
+#ifdef _MSC_VER
+        for (auto& x : section)
+        {
+            auto keywords = handler(x);
+            for (auto& keyword : keywords)
+            {
+                this->keywords.push_back(keyword);
+            }
+        }
+#else
         this->keywords = fun::concat( fun::map( handler, section ) );
```

Comment:
> It's not rocket science, but it's slower, noisy and annoyingly manual and duplicated logic.

but it is still better than having both codepaths and some preprocessor-foo? I personally prefer some minor duplication of trivial code if it makes the code simpler in the sense that it reduces the mental load of stuff which you need to know in order to understand it. (remember the recent 'leftpad' incident?) That's my personal opinion, though.

> map and concat are scheduled to be replaced with lazy implementations too, which should be even more effective. map is seeing some more use throughout parser (and output) and abstractions are nice.

most places where these are used are in no way performance critical, I suppose?

> It's intent is also clearer (in my opinion anyway), and map and concat are pretty well established names. Maybe concat could've been aliased or renamed to flatten, but that's a different discussion.

if we were talking lisp or scheme, I'd agree. But in C++, these are not standard constructs, and people are not used to the functional programming paradigm...